src/utils/TextToSpeech.js

```diff
@@ -0,0 +1,74 @@
+const supportedVoiceLanguages = ['en', 'it'];
+let recommendedVoices = [];
+supportedVoiceLanguages.forEach(supportedVoiceLanguage => {
+  recommendedVoices[
+    supportedVoiceLanguage
+  ] = require(`./textToSpeech/${supportedVoiceLanguage}.json`);
+});
+
+export function hasTextToSpeechVoices() {
+  return speechSynthesis.getVoices().length !== 0;
+}
+
+export function startTextToSpeech(text, locale, onComplete) {
+  const voices = speechSynthesis.getVoices();
+  if (voices.length === 0) {
+    return false;
+  }
+
+  const filteredVoices = filterAvailableVoices(recommendedVoices[locale]);
+  if (filteredVoices.length === 0) {
+    return false;
+  }
+
+  const utterance = new SpeechSynthesisUtterance(text);
+
+  for (const voice of voices) {
+    if (voice.name === filteredVoices[0].name) {
+      utterance.voice = voice;
+      utterance.lang = voice.lang;
+      break;
+    }
+  }
+
+  utterance.onend = onComplete;
+
+  speechSynthesis.cancel();
+
+  speechSynthesis.speak(utterance);
+
+  console.log('Saying:', text);
+
+  return true;
+}
+
+export function stopTextToSpeech() {
+  speechSynthesis.cancel();
+}
+
+// From https://github.com/HadrienGardeur/web-speech-recommended-voices.
+function filterAvailableVoices(jsonData) {
+  if (!jsonData) {
+    return [];
+  }
+
+  const availableVoices = [];
+  const voices = speechSynthesis.getVoices();
+
+  jsonData.voices.forEach(function(voice) {
```

Comment:
Good question.  The voice list should already be ordered.  Excerpts from [here](https://github.com/HadrienGardeur/web-speech-recommended-voices?tab=readme-ov-file#guiding-principles):

> - Each voice list is ordered and meant to provide an optimal listening experience on all browsers/OS/languages covered by this project.
> - But each list also includes default options, to make sure that there's always something reliable to lean on.
> - With these two goals in mind, higher quality voices are listed on top of the list, while lower quality voices or specialized ones are listed at the bottom.

It would be great if they could produce an (opinionated) API as we [discussed](https://github.com/code-dot-org/ml-activities/pull/403/files#r1662992090).  

It might be worth borrowing more from their [demo code](https://github.com/HadrienGardeur/web-speech-recommended-voices/blob/2bbf8705e68d82bc2aea2395079c24f3f4a69dcd/demo/script.js), if we like its final voice sort better.  I'm not quite sure why, yet, but on macOS Chrome their additional sorting puts `Female Google voice (US)` (quality: `high`) at the top of the list, but our subset of the code puts `Daniel (UK)` (quality: `normal`) at the top.  `Daniel (UK)` is higher in the JSON file, possibly because of the potential 14 second playback bug with `Female Google voice (US)`, which I believe I have also observed, so it doesn't seem unreasonable.  Definitely worth further thinking, though.

Reply:
Makes sense, thanks for the info!